internal/dataplane/sendconfig/sendconfig.go

```diff
@@ -288,3 +269,63 @@ func isConflictErr(err error) bool {
 
 	return false
 }
+
+type KonnectAwareClient interface {
+	IsKonnect() bool
+}
+
+type StatusClient interface {
+	Status(context.Context) (*kong.Status, error)
+}
+
+// hasConfigurationChanged verifies whether configuration has changed by comparing old and new config's SHAs.
+// In case the SHAs are equal, it still can return true if a client is considered crashed based on its status.
+func hasConfigurationChanged(
+	ctx context.Context,
+	oldSHA, newSHA []byte,
+	client KonnectAwareClient,
+	statusClient StatusClient,
+	log logrus.FieldLogger,
+) (bool, error) {
+	if !bytes.Equal(oldSHA, newSHA) {
+		return true, nil
+	}
+	if !hasSHAUpdateAlreadyBeenReported(newSHA) {
+		log.Debugf("sha %s has been reported", hex.EncodeToString(newSHA))
+	}
+	// In case of Konnect, we skip further steps as it doesn't report its configuration hash.
+	if client.IsKonnect() {
+		return false, nil
+	}
```

Comment:
But 🤔 when that returns `false` then KIC won't send config to Gateway. Is that desired? Or am I missing something?

Reply:
Here we're already past `newSHA == oldSHA` check, so it's just a matter of not being able to detect whether Konnect has no configuration. We don't expect Konnect to crash and lose the configuration we already sent it.